src/main/java/uk/ac/ic/wlgitbridge/snapshot/base/Request.java

```diff
@@ -159,4 +161,11 @@ private void request(HttpRequest request) throws IOException {
         future = request.executeAsync();
     }
 
+    private void setTimeouts(HttpRequest request) {
+        // timeouts are 20s by default
+        int threeMinutesInMs = 1000 * 60 * 3;
+        request.setConnectTimeout(threeMinutesInMs);
```

Comment:
Do we really need to increase the connect timeout? I would guess that if anything we hit the read timeout while we're waiting for the API to build the snapshot for us. It seems unlikely that it should take 3 minutes for the socket to connect.

Reply:
Yeah, probably not necessary :+1: